elixir/apps/api/lib/api/client/channel.ex

```diff
@@ -306,11 +306,13 @@ defmodule API.Client.Channel do
       with {:ok, resource} <-
              Resources.fetch_and_authorize_resource_by_id(resource_id, socket.assigns.subject),
            {:ok, [_ | _] = gateways} <-
-             Gateways.list_connected_gateways_for_resource(resource, preload: :group),
+             Gateways.all_connected_gateways_for_resource(resource, socket.assigns.subject,
+               preload: :group
+             ),
            gateway_groups = Enum.map(gateways, & &1.group),
            {relay_hosting_type, relay_connection_type} = Gateways.relay_strategy(gateway_groups),
            {:ok, [_ | _] = relays} <-
-             Relays.list_connected_relays_for_resource(resource, relay_hosting_type) do
+             Relays.all_connected_relays_for_resource(resource, relay_hosting_type) do
```

Comment:
Since we have various kinds of returns I've decided to come up with a convention just for us:

- `list_*` returns `{:ok, results, metadata}` - basically a paginated query used everywhere in the UI
- `all_*` returns `{:ok, results}` - used in places where we need to really fetch all the results without pagination (eg. list of enabled providers when we build the sign in form)
- `all_*!` returns `results` or raises an error. This is when we just do the query and there is no need to wrap it in the tuple, mostly used for internal stuff like background jobs